### DISCUSSION TITLE
Changing middleware execution order for middleware added by modules

### DISCUSSION THREAD

**Discussion by chexburger** [nuxt/nuxt#16278] [Questions]
I'm currently using a module for handling sessions. This works great and I can access it just fine when I'm accessing it from a server route. Part of what I want to use these sessions for is done within my middleware, but the module middleware executes after my custom middleware and thus I cannot access anything added to the event context.
I would like to run this module's middleware before all other middleware, but I am unknowledgeable in forcing a middleware order, especially for middleware I didn't create. I saw someone mention Hooks for this without any pointers as to how, but from my research into hooks, I've been unable to find any way to do that.

**Comment by BracketJohn** [comment]
Hey @chexburger! 

Thanks for asking this question - I think what module it may be related to -> I'm also trying to resolve this right now 😉 😄 

One solution that does not work is to use naming, it was a blind shot based on the nuxt-docs on middleware ordering on the app side.

Two similar, other idea that occured to me:
1. Add the middleware inside a `server/plugin` so that the middleware is added at runtime (where nuxt-app middlewares are hopefully already added) and not at module build-time where the nuxt-app middlewares are not yet there. I couldn't quite figure out how to do this yet,
2. Add the middleware in some hook -> the one's I tried were not the right ones though

For (1) my idea was to change `nitroApp` following these lines:
```ts
export default defineNitroPlugin(async (nitroApp) => {
  // something like this, but unsure if the timing here is correct.
  nitroApp.h3App.stack.push({
    handler: theSessionMiddleware
  })
})
```

As this is @pi0's special area, maybe he knows how to work this?
